cpp/src/arrow/compute/kernels/hash_aggregate_test.cc

```diff
@@ -2460,6 +2461,315 @@ TEST(GroupBy, Distinct) {
   }
 }
 
+MATCHER_P2(AnyOfJSON, type, array_json, "") {
```

Comment:
Just reiterating my earlier comment - but we've found before that these convenience macros don't work in all environments.

Reply:
Trying to mitigate this right now.